config-inject/README.md

```diff
@@ -0,0 +1,150 @@
+    naming convention. Config modules have a `config` name export which is an object containing
+    configuration values. The object can be arbitrary nested, but needs to be JSON-serialisable.
+-   `config.dev.mjs` exists and contains configuration used during development
+
+Furthermore, we recommend:
+
+1. Have a `config.ts` file which defines the shape of your config, e.g.:
+
+```ts
+// config/config.ts
+export type AppConfig = {
+    env: 'staging' | 'production' | 'dev'
+    version: string
+    nested: {
+        boi: string
+    }
+}
+```
+
+Then in your per-env config files use use TypeScript via JSDocs annotations, e.g.:
+
+```mjs
+// @ts-check
+/**
+ * @type {import("./config").AppConfig}
+ */
+export const config = {
+    env: 'staging',
+    version: '__INJECTED__',
+    nested: {
+        boi: 'wat'
+    }
+}
+```
+
+2. Consume the config in your app through a module exposing the typed config:
+
+```ts
+// config/index.ts
+import {getConfig} from '@pleo-io/spa-config-inject'
+import type {AppConfig} from './config'
+
+export default getConfig<AppConfig>()
+```
+
+Then in your app code:
+
+```ts
+// some-file.ts
+import config from './config'
+
+let boi = config.nested.boi // string
+```
+
+## Configuration
+
+You can configure the tool by adding a spaConfig namespace to your package.json file. The
+configuration options are as follows:
+
+-   `configDir` (string): The location of your app configuration files.
+-   `buildDir` (string): The location of the HTML template file.
+-   `templateFileName` (string, optional, default: `_index.html`): The name of the template HTML
+    file (relative to build dir).
+-   `outputFileName` (string, optional, default: `index.html`): The name of the output HTML file
+    (relative to build dir).
+-   `devConfigOverrideFile` (string, optional, default: `config.dev.json`): The name of the
+    development config override file.
+
+## Vite plugin
+
+To inject the dev config when running the app locally, add the Vite plugin to your Vite config:
+
+```ts
+import {defineConfig} from 'vite'
+import {inlineDevelopmentConfig} from '@pleo-io/spa-config-inject/vite'
+
+export default defineConfig((config) => {
+    return {
+        plugins: [inlineDevelopmentConfig(config)]
+    }
+})
+```
+
+## Dev overrides
+
+When running the app in dev mode the first time, an empty `config.dev.json` (or another name if you
+configure it via `devConfigOverrideFile` option) will be created at the root of the project. This
+file allows you to temporarily override the dev config (e.g. change an API route) without having to
+remember to undo your changes before committing. Remember to add that file to `.gitingored` as it's
+not supposed to be versioned.
+
+## CLI Usage
+
+In your CD flow you'll need to run the CLI for each environment you deploy to, in order to generate
+the HTML with baked in config for that env.
+
+```sh
+$ pnpm spa-config-inject --env <env>
```

Comment:
The reason why we currently have this script in a nested folder in product-web with it's own lock file, is to avoid installing all dependencies – to make this as fast as possible. 

So I suggest we encourage people to use `npx` (which is better than `pnpx`, as this doesn’t require pnpm to be installed).